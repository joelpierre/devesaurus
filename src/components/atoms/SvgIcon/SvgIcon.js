import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './SvgIcon.module.scss';

const SvgIcon = ({ name, description, fill, className, center }) => {
  const req = require.context('../../../assets/icons/', true, /^\.\/.*\.svg$/);

  const svg = () => {
    let svg = markup();

    const titleReg = /<[title/>][^>]*><\/[title>]+>/;
    const descReg = /<[desc/>][^>]*><\/[desc>]+>/;

    svg = svg.replace(titleReg, `<title>${name}</title>`);
    svg = svg.replace(descReg, `<desc>${description} for ${name}</desc>`);

    return svg;
  };

  const markup = () => {
    return req('./' + name + '.svg').replace(
      /^<svg /,
      `<svg role="img" class="${classNames([
        styles.icon__svg,
      ])}" style="fill:${fill};"`
    );
  };

  return (
    <span
      data-test="component-svg-icon"
      aria-label={name}
      className={classNames([
        styles.icon,
        { [styles.icon__center]: center },
        className,
      ])}
      dangerouslySetInnerHTML={{ __html: svg() }}
    />
  );
};

SvgIcon.defaultProps = {
  fill: `currentColor`,
  description: 'An SVG Icon',
  className: null,
  center: false,
};

SvgIcon.propTypes = {
  center: PropTypes.bool,
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  className: PropTypes.string,
  fill: PropTypes.string,
};

export default SvgIcon;
