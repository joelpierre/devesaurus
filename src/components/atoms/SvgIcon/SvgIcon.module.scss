.icon {
  display: inline-flex;
  position: relative;
  min-height: .8rem;
  max-width: 100%;
  height: 100%;
  margin: 0;
  box-sizing: content-box;

  &__svg {
    width: inherit;
    height: inherit;
  }

  &__center {
    width: 100%;
    margin: 0 auto;
  }
}
