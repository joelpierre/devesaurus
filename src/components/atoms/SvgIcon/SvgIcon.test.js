import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SvgIcon from './SvgIcon';

configure({ adapter: new Adapter() });

describe('<SvgIcon/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SvgIcon classes={`test-class`} name={`codepen`}/>,
    );
  });

  it('Should render SvgIcon Component', () => {
    expect(wrapper)
      .toMatchSnapshot();
  });

});
