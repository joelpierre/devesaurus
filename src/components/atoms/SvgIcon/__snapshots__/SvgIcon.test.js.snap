// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon/> Should render SvgIcon Component 1`] = `ShallowWrapper {}`;
